contrib/xz-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,246 @@
+        ${LIBRARY_DIR}/liblzma/api/lzma/delta.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/filter.h
+        ${LIBRARY_DIR}/liblzma/api/lzma.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/hardware.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/index.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/index_hash.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/lzma12.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/stream_flags.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/version.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/vli.h
+        ${LIBRARY_DIR}/liblzma/check/check.h
+        ${LIBRARY_DIR}/liblzma/check/crc32_table_be.h
+        ${LIBRARY_DIR}/liblzma/check/crc32_table_le.h
+        ${LIBRARY_DIR}/liblzma/check/crc64_table_be.h
+        ${LIBRARY_DIR}/liblzma/check/crc64_table_le.h
+        ${LIBRARY_DIR}/liblzma/check/crc_macros.h
+        ${LIBRARY_DIR}/liblzma/common/alone_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/block_buffer_encoder.h
+        ${LIBRARY_DIR}/liblzma/common/block_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/block_encoder.h
+        ${LIBRARY_DIR}/liblzma/common/common.h
+        ${LIBRARY_DIR}/liblzma/common/easy_preset.h
+        ${LIBRARY_DIR}/liblzma/common/filter_common.h
+        ${LIBRARY_DIR}/liblzma/common/filter_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/filter_encoder.h
+        ${LIBRARY_DIR}/liblzma/common/index_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/index_encoder.h
+        ${LIBRARY_DIR}/liblzma/common/index.h
+        ${LIBRARY_DIR}/liblzma/common/memcmplen.h
+        ${LIBRARY_DIR}/liblzma/common/outqueue.h
+        ${LIBRARY_DIR}/liblzma/common/stream_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/stream_flags_common.h
+        ${LIBRARY_DIR}/liblzma/delta/delta_common.h
+        ${LIBRARY_DIR}/liblzma/delta/delta_decoder.h
+        ${LIBRARY_DIR}/liblzma/delta/delta_encoder.h
+        ${LIBRARY_DIR}/liblzma/delta/delta_private.h
+        ${LIBRARY_DIR}/liblzma/lz/lz_decoder.h
+        ${LIBRARY_DIR}/liblzma/lz/lz_encoder.h
+        ${LIBRARY_DIR}/liblzma/lz/lz_encoder_hash.h
+        ${LIBRARY_DIR}/liblzma/lz/lz_encoder_hash_table.h
+        ${LIBRARY_DIR}/liblzma/lzma/fastpos.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma2_decoder.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma2_encoder.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma_common.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma_decoder.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma_encoder.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma_encoder_private.h
+        ${LIBRARY_DIR}/liblzma/rangecoder/price.h
+        ${LIBRARY_DIR}/liblzma/rangecoder/range_common.h
+        ${LIBRARY_DIR}/liblzma/rangecoder/range_decoder.h
+        ${LIBRARY_DIR}/liblzma/rangecoder/range_encoder.h
+        ${LIBRARY_DIR}/liblzma/simple/simple_coder.h
+        ${LIBRARY_DIR}/liblzma/simple/simple_decoder.h
+        ${LIBRARY_DIR}/liblzma/simple/simple_encoder.h
+        ${LIBRARY_DIR}/liblzma/simple/simple_private.h
+)
+
+ADD_LIBRARY(liblzma ${Sources} ${Headers})
```

Comment:
But there is no users of this library, I guess that is the reason for an error:

```
/home/centos/ClickHouse/src/IO/LzmaReadBuffer.h:7:10: fatal error: lzma.h: No such file or directory
7 | #include <lzma.h>
```

Reply:
I guess it should be added in the same manner as here - https://github.com/ClickHouse/ClickHouse/blob/90dcc135b4043ecca726adc2dc9470ef48a0fba2/src/CMakeLists.txt#L325-L330

Reply:
Did it in similar fashion